pyproject.toml

```diff
@@ -31,6 +31,9 @@ classifiers = [
 requires = ["pex>=1.1.15"]
 requires-python = ">=2.7,<3.10,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,!=3.5.*"
 
+[tool.flit.metadata.requires-extra]
+test-gcp-http = ["flask"]
```

Comment:
I think this will need to be a 2 or 3 element list with environment markers that split the flask dep for Python 2.7 (maybe 3.5) and >=3.6. CI failed trying to use 2.0.1 for Python 2.7 but release notes say the req should be `flask<2; python_version == '2.7'`: https://flask.palletsprojects.com/en/2.0.x/changes/#version-2-0-0

Reply:
If I'm specifying the flask requirement in the example that I'm running, will I also need it in lambdex's requirements as an extra?

Reply:
Yes. You can imagine a user has a PEX file to lambdex and nothing else. If they want to use `lambdex test` and their gcp-http event handler uses duck typing and has no explicit Flask imports and thus no Flask dependency embedded in the PEX, then this import goes boom:
https://github.com/pantsbuild/lambdex/pull/22/files#diff-4ff255eb08b5673f8ce58b8260fe52c4004b235e6cbef0f954d02b752f934a47R215

(I could not find any mention of `flask.Request` being the guaranteed even object handed to the http event handler but all examples comment / doc this so I'm guessing it is, at least in practice.)

Reply:
Keep in mind, the extras do not suffer the problem of the examples/gcp_http/requirements.txt above, The extras can be expected to be installed by a capable resolver - the Pip family (includes modern Pex) and Poetry.